File: material/src/main/scala/material/plugin/S3PackageRepositoryMaterial.scala
package material.plugin

import com.thoughtworks.go.plugin.api.material.packagerepository.{PackageMaterialPoller, PackageMaterialConfiguration, PackageMaterialProvider}
import material.plugin.config.S3PackageMaterialConfiguration


class S3PackageRepositoryMaterial extends PackageMaterialProvider {
  override def getConfig: PackageMaterialConfiguration = new S3PackageMaterialConfiguration()
  override def getPoller: PackageMaterialPoller = new S3PackageMaterialPoller()
}

File: material/src/main/java/com/indix/gocd/s3material/plugin/S3PackageRepositoryMaterial.java
package com.indix.gocd.s3material.plugin;

import com.indix.gocd.s3material.config.S3PackageMaterialConfiguration;
import com.thoughtworks.go.plugin.api.annotation.Extension;
import com.thoughtworks.go.plugin.api.material.packagerepository.PackageMaterialConfiguration;
import com.thoughtworks.go.plugin.api.material.packagerepository.PackageMaterialPoller;
import com.thoughtworks.go.plugin.api.material.packagerepository.PackageMaterialProvider;

@Extension
public class S3PackageRepositoryMaterial implements PackageMaterialProvider {
    @Override
    public PackageMaterialConfiguration getConfig() {
        return new S3PackageMaterialConfiguration();
    }

    @Override
    public PackageMaterialPoller getPoller() {
        return new S3PackageMaterialPoller();
    }
}
